src/metadata/Metadata.ts

```diff
@@ -193,11 +197,17 @@ export class Metadata extends MarketplaceApi {
      * @returns DDO of the asset.
      */
     public async retrieveDDO(
-        did: DID | string,
+        did: DID | string | undefined,
```

Comment:
instead of undefined maybe we can just make did optional

Reply:
changed